app/models/reservation.rb

```diff
@@ -159,6 +168,10 @@ def assign_actuals_off_reserve
     self.actual_end_at   ||= reserve_end_at
   end
 
+  def assign_category_attribute(user, params)
+    assign_attributes(category: params[:category]) if user.operator_of?(product.facility)
```

Comment:
When updating an admin reservation (everything worked fine for creation), I'm getting an error here because `params` is `nil`.